### ISSUE TITLE
Multiplane Frame Support

### ISSUE THREAD

**Issue by jczorch** [mikeboers/PyAV#206]
Will you ever add in multiplane frame support? Currently I am stuck on not being able to generate a PIL Image from a multiplane frame.

**Comment by mikeboers** [comment]
You mean `VideoFrame.planes` ?

```
>>> import av
>>> fh = av.open('/path/to/your.mov')
>>> v = fh.streams.get(video=0)[0]
>>> f = None
>>> while not f:
...     p = next(fh.demux(v))
...     f = p.decode()
>>> f
[<av.VideoFrame #0, yuv420p 1920x1080 at 0x10fb07d00>]
>>> f[0].planes
(<av.VideoPlane at 0x10fb80d08>, <av.VideoPlane at 0x10fb80d60>, <av.VideoPlane at 0x10fb80db8>)
```

**Comment by jczorch** [comment]
Correct. So now if you wanted to generate a numpy array from that frame you will raise an exception:
>>> f[0].to_nd_array()
ValueError: Cannot conveniently get numpy array from multiplane frame

And if you do f[0].to_image() you will get a Segmentation fault: 11

**Comment by mikeboers** [comment]
I don't think there is an obvious Numpy layout for all multi-plane frames; they usually have different dimensions in the planes.

I'm a little surprised by the segfault. Works fine on a source I have over here:

```
>>> f[0]
<av.VideoFrame #0, yuv420p 1920x1080 at 0x10510bd00>
>>> f[0].to_image()
<PIL.Image.Image image mode=RGB size=1920x1080 at 0x10518AF10>
```

Could you please upload a small sample to: https://www.dropbox.com/request/DdzUvAReKIHWgXIn0vqd

**Comment by mikeboers** [comment]
I'm going to assume the segfault is fixed in the master unless I get a sample. Until then, this is just a docs request.

**Comment by slhck** [comment]
So, when I have a frame, how do I get its Y component plane as a numpy array?

```
(Pdb) frame.to_nd_array()
*** ValueError: Cannot conveniently get numpy array from multiplane frame
(Pdb) frame.planes
(<av.VideoPlane at 0x112d230e8>, <av.VideoPlane at 0x112d23228>, <av.VideoPlane at 0x112d23318>)
(Pdb) frame.planes[0].to_nd_array()
*** AttributeError: 'av.video.plane.VideoPlane' object has no attribute 'to_nd_array'
```

**Comment by slhck** [comment]
I guess

```
np.frombuffer(frame.planes[0], np.uint8).reshape(frame.height, frame.width)
```

might work?

**Comment by jlaine** [comment]
As far as ndarray is concerned, I think this has been completely addressed in #415

**Comment by mikeboers** [comment]
Agreed.
